genet/core.py

```diff
@@ -657,23 +657,28 @@ def _find_link_ids_on_s2_geometry(self, gdf, how, s2_input):
         else:
             raise NotImplementedError('Only `intersect` and `within` options for `how` param.')
 
-    def add_node(self, node: Union[str, int], attribs: dict = None, silent: bool = False):
+    def add_node(self, node: Union[str, int], attribs: dict, silent: bool = False):
         """
         Adds a node.
         :param node:
-        :param attribs: should include spatial information x,y in epsg cosistent with the network or lat lon in
-        epsg:4326
+        :param attribs: must include spatial information x,y in epsg cosistent with the network, or lat lon in epsg:4326
         :param silent: whether to mute stdout logging messages
         :return:
         """
-        if attribs is not None:
-            self.graph.add_node(node, **attribs)
-        else:
-            # TODO do not add nodes without spatial info
-            self.graph.add_node(node)
+        # check for spatial info
+        if ('lat' not in attribs.keys() and 'lon' not in attribs.keys()) and \
+                ('x' not in attribs.keys() and 'y' not in attribs.keys()):
```

Comment:
this statement is a bit crazy :D it's difficult to understand with all the ands and nots (btw, to check for a key in dict you dont need the `.keys()`). I think you should write this differently so it's more readable:
```python
keys = set(attribs.keys())
if not ({'lat', 'lon'}.issubset(keys) or {'x', 'y'}.issubset(keys)):
    raise ..

```